server/src/transactions/dto/create-transaction.dto.ts

```diff
@@ -0,0 +1,29 @@
+import { IsNotEmpty, IsEnum, IsOptional } from 'class-validator';
+import { TransactionStatus } from '../transaction-status.enum';
+
+import type { User } from 'src/users/entities/user.entity';
+import type { Organization } from 'src/organizations/entities/organization.entity';
+import type { Asset } from 'src/assets/entities/asset.entity';
+
+export class CreateTransactionDto {
+  @IsNotEmpty()
+  donater_user: User;
```

Comment:
### Fix:
Should be:
``` typescript
 @IsNotEmpty()
  donater_user_id: number;
```

Reply:
hmmm, are you sure?
would nest know to add a `.donater` attr or function to the transation model then? now that i think about it, idk how to control the model in nest, like if i wanted to write a custom function like `asset.close()` (that updates the status to all the transactions related to that asset, how would i do that?

Reply:
Looking at your example, 

This:
```bash
curl -X POST localhost:3001/api/assets \
-H "Content-Type: application/json" \
-d '{ "title": "thing 1", "description": "its a thing", "type": "donation", "quantity": "1",
"poster": { "first_name":"test", "last_name":"user", "email":"test@test.com", "id": "18" } }'
```

can be replaced by example blow, if you change the createDto to use poster_id instead of a whole new poster object. The return type will be whatever the repository returns (`transactionEntity`):
```bash
curl -X POST localhost:3001/api/assets \
-H "Content-Type: application/json" \
-d '{ "title": "thing 1", "description": "its a thing", "type": "donation", "quantity": "1",
"poster_id": 18 }'
```

Reply:
Then, you would have to implement something like: https://github.com/Nonprofit-Exchange-Hub/web-app/compare/create-transactions-resource...sample/create-transactions-resource

Also, if you are going to stick with the `getTransactionsDto` as the get all filtering model, there is some funky formatting we have to do in the front end to send the nested object in the url query params. See [this](https://nickdrane.com/build-your-own-nested-query-string-encoder/)